C4/AuthExtra.pm

```diff
@@ -0,0 +1,81 @@
+package C4::AuthExtra;
+
+=expl
+get_timeout returns timeout based on user's category
+value is in sysprefs variable BorrowerCategoryTimeout
+value is in yaml-format example:
+
+---
+USERCATEGORY1: 600
+AUTOMAT: 31536000
+
+=cut
+
+use C4::Context;
+use Try::Tiny;
+use YAML;
+
+##################################
+#gets timeout based on borrower's category
+sub get_timeout {
+    my $error;
+    my ($cardnumber,$timeout) = @_;
+    
+    my $dbh   = C4::Context->dbh;
+    my $valueyaml;
+    my $retval = $timeout;
+ 
+    try {
+
+        #get catgorycode
```

Comment:
Instead of commenting there could be a function called get_categorycode like you have get_timeout.